src/Data/String/Regex.js

```diff
@@ -29,7 +29,9 @@ exports.flags = function (r) {
 
 exports.test = function (r) {
   return function (s) {
-    return r.test(s);
+    var result = r.test(s);
+    r.lastIndex = 0;
```

Comment:
In the Purescript Regex API, there is no other function that affects `lastIndex` (if I'm correct), so I thought resetting to `0` would be fine. But your solution is the safer alternative (there might be foreign code which affects the `lastIndex` property). Thanks.